site/content/docs/howto/configure-supervisor-with-workspace_one_access.md

```diff
@@ -0,0 +1,153 @@
+1. Create a new app:
+   1. Click `New`.
+   1. Enter a name for your app, such as "My Kubernetes Clusters".
+   1. For `Authentication Type`, select `OpenID Connect`.
+   1. Enter the Target URL. This value is required but unused and may be set to the `spec.issuer` you configured in your `FederationDomain`.
+   1. Enter the Redirect URL. This is the `spec.issuer` you configured in your `FederationDomain` appended with `/callback`.
+   1. Enter the Client ID to a value such as "pinniped-supervisor" (this cannot be changed later).
+   1. Enter the Client Secret. This should be set to secure value such as the output of `openssl rand -hex 32`.
+   1. Set both "Open in Workspace ONE Web" and "Show in User Portal" options to "No"
+   1. Set the desired Access Policies for the app, such as requiring smart card login.
+   1. Save and assign the app to the desired users and/or groups. This can be used to restrict which users can log in to Kubernetes using this integration.
+1. Configure the token TTLs and scopes. Navigate to _Catalog_ > _Settings_ > _Remote App access_ and click on the "pinniped-supervisor" client.
+   1. Edit the _Client Configuration_:
+      1. `Issue Refresh Token` must be checked
+      1. Set `Access Token Time-To-Live (TTL)` to 5 minutes
+      1. Set `Refresh Token Time-To-Live (TTL)` to 9 hours (or shorter if you wish to require more frequent logins)
+      1. Set `Idle Token Time-to-Live (TTL)` to 9 hours (or shorter if you wish to enforce an inactivity timeout)
+   1. Edit the _Scope_ configuration:
+      1. `OpenID` must be checked
+      1. Check `Email` if you plan to use email as the username claim
+      1. Check `Group` if you plan to use groups in your Kubernetes environment
+      1. Uncheck all other scopes
+
+## Configure the Supervisor
+
+Create an [OIDCIdentityProvider](https://github.com/vmware-tanzu/pinniped/blob/main/generated/{{< latestcodegenversion >}}/README.adoc#oidcidentityprovider) in the same namespace as the Supervisor.
+
+For example, this OIDCIdentityProvider and corresponding Secret use Workspace ONE Access's `email` claim as the Kubernetes username:
+
+```yaml
+apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+kind: OIDCIdentityProvider
+metadata:
+  namespace: pinniped-supervisor
+  name: ws1
+spec:
+
+  # Specify the issuer URL (no trailing slash). Change this to be the
+  # actual issuer of your Workspace ONE Access environment.  Note that
+  # the Workspace ONE Access issuer ends with the string "/SAAS/auth."
+  issuer: https://ws1.my-company.com/SAAS/auth
+
+  # Specify how to form authorization requests to Workspace ONE Access.
+  authorizationConfig:
+
+    # Request any scopes other than "openid" for claims besides
+    # the default claims in your token. The "openid" scope is always
+    # included.
+    #
+    # See the example claims below to learn how to customize the claims returned.
+    additionalScopes: [group, email]
```

Comment:
Just confirming, users don't need to do anything to ask for refresh tokens, e.g. request `offline_access`?

Reply:
Yeah, here is an excerpt from the session data:

```json
{
  "providerUID": "6f326005-ffc3-4568-b316-8442169eb637",
  "providerName": "ws1",
  "providerType": "oidc",
  "warnings": null,
  "oidc": {
    "upstreamRefreshToken": "NmFhNTdkMTktMmI...Y0ODE0Nzg3Nw",
    "upstreamAccessToken": "",
    "upstreamSubject": "td.mon...5C",
    "upstreamIssuer": "https://td-mo...w.com/SAAS/auth"
  }
}
```